src/main/java/com/skelril/skree/content/teleport/BringCommand.java

```diff
@@ -55,9 +55,30 @@ public CommandResult execute(CommandSource src, CommandContext args) throws Comm
 
         Optional<GameModeData> data = target.getData(GameModeData.class);
         if (!data.isPresent() || !(data.get().getGameMode() == GameModes.CREATIVE || data.get().getGameMode() == GameModes.SPECTATOR)) {
+            //make sure no lava/fire is at the targets destination
+            dest = dest.add(0, 2, 0);
+            while (dest.getFloorY() > 0) {
+                dest = dest.add(0, -1, 0);
+
+                if (!(targetExtent.getBlock(dest.toInt()).getType().equals(BlockTypes.AIR))) {
+                    if (targetExtent.getBlock(dest.toInt()).getType().equals(BlockTypes.LAVA) || targetExtent.getBlock(dest.toInt()).getType().equals(BlockTypes.FLOWING_LAVA) || targetExtent.getBlock(dest.toInt()).getType().equals(BlockTypes.FIRE)) {
+                        src.sendMessage(Texts.of(TextColors.RED, "My lord, thou would burn thy feet teleporting here!"));
+                        return CommandResult.empty();
+                    }
+                    break;
+                }
+            }
+            dest = ((Player) src).getLocation().getPosition();
+
+            //make sure target doesn't fall
             while (dest.getFloorY() > 0 && targetExtent.getBlock(dest.toInt()).getType().equals(BlockTypes.AIR)) {
                 dest = dest.add(0, -1, 0);
             }
+
+            //make sure target doesn't suffocate
+            while (!(targetExtent.getBlock(dest.toInt()).getType().equals(BlockTypes.AIR) && dest.getFloorY() < 255)) {
+                dest = dest.add(0, 1, 0);
+            }
```

Comment:
The other thing is, if you look at this loop, it seems that this will end up placing you 1 block above. The fall protection code, actually finds the first non-air block. Then as you can see below it adds "dest.add(0, 1, 0)" to put the player back in the right spot. So, this should be checking dest.toInt().add(0, 1, 0) (and possibly dest.toInt().add(0, 2, 0).

If you wanted to be efficient about it you could do something like the following:
````java
while (dest.getFloorY() < 255) {
    if (!targetExtent.getBlock(dest.toInt().add(0, 2, 0).getType().equals(BlockTypes.AIR)) {
        dest = dest.add(0, 2, 0);
        continue;
    }
    if (!targetExtent.getBlock(dest.toInt().add(0, 1, 0).getType().equals(BlockTypes.AIR)) {
        dest = dest.add(0, 1, 0);
        continue;
    }
    break;
}
````

This is slightly more efficient as when you know that they aren't going to have air at their head, you can skip 2 blocks, since that's the minimum required to find a new air block at their head.

If you use this, you should be able to remove the dest.add(0, 1, 0) line below.